src/indexes.ts

```diff
@@ -302,30 +305,46 @@ class Index<T extends Record<string, any> = Record<string, any>> {
   ///
 
   /**
-   * Get documents of an index
+   * Get documents of an index.
    *
-   * @param parameters - Parameters to browse the documents
-   * @returns Promise containing Document responses
+   * @param parameters - Parameters to browse the documents. Parameters can
+   *   contain the `filter` field only available in Meilisearch v1.2 and newer
+   * @returns Promise containing the returned documents
    */
   async getDocuments<D extends Record<string, any> = T>(
     parameters: DocumentsQuery<D> = {}
   ): Promise<ResourceResults<D[]>> {
-    const url = `indexes/${this.uid}/documents`
-
-    const fields = (() => {
-      if (Array.isArray(parameters?.fields)) {
-        return parameters?.fields?.join(',')
+    parameters = removeUndefinedFromObject(parameters)
+
+    // In case `filter` is provided, use `POST /documents/fetch`
+    if (parameters.filter || typeof parameters.filter === 'string') {
```

Comment:
```suggestion
    if (parameters.filter || typeof parameters.filter === 'string') {
```
I think we should check for an array, here, no?

Reply:
I was checking the tests and passing the array works. But I'm curious to know which conditional is catching with this validation.

Reply:
an empty string in javascript is a `falsy` by adding a condition to check if it is a string I also include `""`. Nonetheless, I could change this condition to just
```
parameters.filter !== undefined
```